text/0000-factoryFor.md

```diff
@@ -0,0 +1,452 @@
+
+```
+                    Current:
+       +-------------------------------+
+       |                               |
+       |      /app/models/foo.js       |
+       |                               |
+       +-------------------------------+
+                       |
+                       |
+                       |
+                       v
+            +--------------------+
+            |  Class[model/Foo]  |
+            +--------------------+
+                       |
+                       |
+                       |
+       first test run  |   nth test run
+           +-----------+----------+
+           |                      |
+           |                      |
+           |                      |
+           v                      v
++--------------------+ +--------------------+
+|     subclass of    | |     subclass of    |
+|  Class[model/Foo]  | |  Class[model/Foo]  |
++--------------------+ +--------------------+
+```
+
+This means, between test runs 2 instances of `model:foo` will have a common
+shared ancestor the grandparent `Class[model/Foo]`.
+
+This implementation of `factoryFor` proposes to remove the intermediate
+subclass and instead have a generic
+factory object which holds the injections and allows for injected instances
+to be created. The resulting object graph would look something like this:
+(between test runs the 2 instances of `model:foo` will have a shared ancestor
+(parent) `Class[model/Foo]`.
+
+```
+                  Proposed:
+      +-------------------------------+
+      |                               |
+      |      /app/models/foo.js       |
+      |                               |
+      +-------------------------------+
+                      |
+                      |
+                      |
+                      v
+           +--------------------+
+           |  Class[model/Foo]  |
+           +--------------------+
+                      |
+                      |
+                      |
+      first test run  |   nth test run
+           +----------+-----------+
+           |                      |
+           |                      |
+           |                      |
+           v                      v
++--------------------+ +--------------------+
+|     Factory of     | |     Factory of     |
+|  Class[model/Foo]  | |  Class[model/Foo]  |
++--------------------+ +--------------------+
+```
+
+With `factoryFor` instances of `model:foo` will share a common constructor.
+Any state stored on the constructor would of course leak between the tests.
+
+An example implementation of `factoryFor` can be reviewed [on this GitHub
+comment](https://github.com/emberjs/rfcs/issues/125#issuecomment-193827658).
+
+##### Implications for `owner.register`
+
+Currently, factories registered into Ember's DI system are required to
+provide an `extend` method. Removing support for extend-based DI in `_lookupFactory`
+will permit factories without `extend` to be registered. Instead factories
```

Comment:
I agree with this change, but it is likely the second "breaking change" in this RFC. Is this a reasonable concern, or is it ok, and if so how do we migrate forwards + minimize any potential fallout.